### ISSUE TITLE
Add temperature scaling to attention_forward (configurable via AttentionOpts.temperature)

### ISSUE THREAD

**Issue by sreehariam** [sreehariam/fa-research-cpp#8]
### Summary
Add temperature scaling to attention computation: divide logits by `opts.temperature`
before applying softmax. Lower temperatures (e.g., 0.5) make distributions sharper,
higher temperatures (>1) make them smoother.

### Details
Extend `AttentionOpts` in `include/fa/types.hpp`:

```
struct AttentionOpts {
    bool causal = false;
    float temperature = 1.0f;   // new field
};
```

Update attention_forward in src/attention_ref.cpp:

Before softmax, scale logits by dividing with opts.temperature (if != 1).

Handle invalid cases (temperature <= 0) with std::invalid_argument.

**Acceptance**
- temperature == 1.0f → identical to current behavior (baseline P2P).
- temperature < 1 → sharper softmax (rows approach one-hot).
- temperature > 1 → smoother distribution (flatter output).
- Temperature > 1.0 (softer) vs baseline (sharper)
- All prior tests remain passing (backward-compatible).
- Add 3 new F2P tests verifying scaling effect.
